File: core/src/nl/t64/game/rpg/screens/shop/ShopSlot.java
package nl.t64.game.rpg.screens.shop;

import com.badlogic.gdx.graphics.Color;
import nl.t64.game.rpg.GameData;
import nl.t64.game.rpg.Utils;
import nl.t64.game.rpg.components.party.InventoryContainer;
import nl.t64.game.rpg.components.party.InventoryGroup;
import nl.t64.game.rpg.components.party.SkillItemId;
import nl.t64.game.rpg.screens.inventory.inventoryslot.InventorySlot;
import nl.t64.game.rpg.screens.inventory.itemslot.InventoryImage;
import nl.t64.game.rpg.screens.inventory.tooltip.ItemSlotTooltip;


class ShopSlot extends InventorySlot {

    ShopSlot(int index, ItemSlotTooltip tooltip, InventoryContainer inventory) {
        super(index, InventoryGroup.SHOP_ITEM, tooltip, inventory);
    }

    void refreshPurchaseColor() {
        getPossibleInventoryImage().ifPresent(this::refreshPurchaseColor);
    }

    private void refreshPurchaseColor(InventoryImage image) {
        GameData gameData = Utils.getGameData();
        int totalMerchant = gameData.getParty().getSumOfSkill(SkillItemId.MERCHANT);
        int costsToBuy = image.inventoryItem.getBuyPricePiece(totalMerchant);
        if (gameData.getInventory().hasEnoughOfItem("gold", costsToBuy)) {
            image.setColor(Color.WHITE);
        } else {
            image.setColor(Color.DARK_GRAY);
        }
    }

}

File: core/src/nl/t64/game/rpg/screens/shop/ShopSlot.kt
package nl.t64.game.rpg.screens.shop

import com.badlogic.gdx.graphics.Color
import nl.t64.game.rpg.Utils.gameData
import nl.t64.game.rpg.components.party.inventory.InventoryContainer
import nl.t64.game.rpg.components.party.inventory.InventoryGroup
import nl.t64.game.rpg.components.party.skills.SkillItemId
import nl.t64.game.rpg.screens.inventory.inventoryslot.InventorySlot
import nl.t64.game.rpg.screens.inventory.itemslot.InventoryImage
import nl.t64.game.rpg.screens.inventory.tooltip.ItemSlotTooltip


internal class ShopSlot(index: Int, tooltip: ItemSlotTooltip, inventory: InventoryContainer
) : InventorySlot(index, InventoryGroup.SHOP_ITEM, tooltip, inventory) {

    fun refreshPurchaseColor() {
        getPossibleInventoryImage()?.let { refreshPurchaseColor(it) }
    }

    private fun refreshPurchaseColor(image: InventoryImage) {
        val totalMerchant = gameData.party.getSumOfSkill(SkillItemId.MERCHANT)
        val costsToBuy = image.inventoryItem.getBuyPricePiece(totalMerchant)
        if (gameData.inventory.hasEnoughOfItem("gold", costsToBuy)) {
            image.color = Color.WHITE
        } else {
            image.color = Color.DARK_GRAY
        }
    }

}
